src/components/Post/Detail/PostImages.tsx

```diff
@@ -41,72 +57,81 @@ export function PostImages({
-                      />
-                    )}
-                  </RoutedDialogLink>
-                  <Reactions
-                    p={4}
-                    className={classes.reactions}
-                    entityId={image.id}
-                    entityType="image"
-                    reactions={image.reactions}
-                    metrics={{
-                      likeCount: image.stats?.likeCountAllTime,
-                      dislikeCount: image.stats?.dislikeCountAllTime,
-                      heartCount: image.stats?.heartCountAllTime,
-                      laughCount: image.stats?.laughCountAllTime,
-                      cryCount: image.stats?.cryCountAllTime,
-                    }}
-                    targetUserId={image.user.id}
-                    readonly={!safe}
-                  />
-                </>
-              )}
-            </ImageGuard2>
-          </Paper>
+          <Fragment key={image.id}>
+            <PostContestCollectionInfoAlert
+              isOwner={isOwner}
+              collectionItem={imageCollectionItem}
+              itemLabel="image"
+            />
+            <Paper
+              key={image.id}
+              radius="md"
+              className="relative overflow-hidden"
+              shadow="md"
+              mx="auto"
+              style={{
+                maxWidth: '100%',
+                width: width < maxWidth ? width : maxWidth,
+                aspectRatio:
+                  image.width && image.height ? `${image.width}/${image.height}` : undefined,
+              }}
+            >
+              <ImageGuard2 image={image} connectType="post" connectId={postId}>
+                {(safe) => (
+                  <>
+                    <ImageGuard2.BlurToggle className="absolute left-2 top-2 z-10" />
+                    <ImageContextMenu image={image} className="absolute right-2 top-2 z-10" />
+                    <RoutedDialogLink name="imageDetail" state={{ imageId: image.id, images }}>
+                      {!safe ? (
+                        <div
+                          style={{
+                            position: 'absolute',
+                            top: 0,
+                            bottom: 0,
+                            aspectRatio: (image.width ?? 0) / (image.height ?? 0),
+                          }}
+                        >
+                          <MediaHash {...image} />
+                        </div>
+                      ) : (
+                        <EdgeMedia
+                          src={image.url}
+                          name={image.name}
+                          alt={
+                            image.meta
+                              ? truncate(image.meta.prompt, { length: constants.altTruncateLength })
+                              : image.name ?? undefined
+                          }
+                          type={image.type}
+                          width={width < maxWidth ? width : maxWidth}
```

Comment:
Let's do `original` for video
```suggestion
                          width={width < maxWidth ? width : maxWidth}
                          original={image.type === 'video'}
```